src/monitor/monitor.c

```diff
@@ -261,7 +263,12 @@ monitor_sbus_RegisterService(TALLOC_CTX *mem_ctx,
     }
 
     /* Fill in svc structure with connection data */
-    svc->conn = sbus_req->conn;
+    svc->conn = sss_ptr_hash_lookup(mt_ctx->sbus_server->names, sbus_req->sender->name, struct sbus_connection);
+    if (svc->conn == NULL) {
+        /* Can this even happen?
```

Comment:
If yes, then there is a bug. I agree with protection, but can you add a debug message like "Bug: connection was not found"?

Reply:
I added a message (`SSSDBG_CRITICAL_FAILURE` as I saw done in other places) and switched to return an error to kill the connection as it is done in the check right before this one.